In [5]:
%matplotlib inline
import numpy as np
import scipy as sp
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

In [6]:
brooklyn=pd.read_csv('brooklyn.csv', low_memory=False)
queens=pd.read_csv('queens.csv', low_memory=False)
brooklyn_homes=pd.read_csv('brooklyn_homes.csv')
queens_homes=pd.read_csv('queens_homes.csv')
queens_low_val=pd.read_csv('queens_low_val.csv')
brooklyn_low_val=pd.read_csv('brooklyn_low_val.csv')
queens=queens.dropna()
brooklyn=brooklyn.dropna()
queens=queens.replace(' -   ','')
queens=queens.replace(' -   ','')

In [7]:
Xqlr, Xqtestlr, yqlr, yqtestlr=train_test_split(queens[['sale_price_clean','year']].values, queens['neighborhood'].values, random_state=5)

In [8]:
clf=LogisticRegression()
clf.fit(Xqlr, yqlr)
print('Accuracy Score For Queens:', accuracy_score(clf.predict(Xqtestlr), yqtestlr))

Accuracy Score For Queens: 0.08261168384879725


This classification model is only 8.26% accurate in predicting the neighborhood in which a sale took place.

In [9]:
Xblr, Xbtestlr, yblr, ybtestlr=train_test_split(brooklyn[['sale_price_clean','year']].values, brooklyn['neighborhood'].values, random_state=5)

In [10]:
clf.fit(Xblr, yblr)
print('Accuracy Score for Brooklyn:', accuracy_score(clf.predict(Xbtestlr), ybtestlr))

Accuracy Score for Brooklyn: 0.09866848753840901


The accuracy score for Brooklyn is marginally better at 9.86% but that is still not acceptable.  Perhaps a model to predict the class of building sold could be more accurate

In [11]:
Xqlr2, Xqtestlr2, yqlr2, yqtestlr2=train_test_split(queens[['sale_price_clean','year']].values, queens['building_class_category'].values, random_state=5)

In [12]:
clf.fit(Xqlr2, yqlr2)
print('Accuracy Score for Queens Building Category:', accuracy_score(clf.predict(Xqtestlr), yqtestlr2))

Accuracy Score for Queens Building Category: 0.518487972508591


Although the score is much improved from the Neighborhood model (51.84% correct), it is still almost a 50-50 chance of getting the building category correct.

In [13]:
Xblr2, Xbtestlr2, yblr2, ybtestlr2=train_test_split(brooklyn[['sale_price_clean','year']].values, brooklyn['building_class_category'].values, random_state=5)

In [14]:
clf.fit(Xblr2, yblr2)
print('Accuracy Score for Brooklyn Building Category:', accuracy_score(clf.predict(Xbtestlr), ybtestlr2))

Accuracy Score for Brooklyn Building Category: 0.4204506657562308


The accuracy score for Brooklyn shows that the model is not as good at predicting building categories in Brooklyn, as it is in Queens.  42.04% is not a good accuracy score but it is better than 9.86% accuracy score for predicting Neighborhoods in Brooklyn.

In [15]:
Xqlr3, Xqtestlr3, yqlr3, yqtestlr3=train_test_split(queens[['sale_price_clean','gross_square_feet_c']].values, queens['year'].values, random_state=5)

In [16]:
clf.fit(Xqlr3, yqlr3)
print('Accuracy Score for Year of Sale In Queens:', accuracy_score(clf.predict(Xqtestlr3), yqtestlr3))

Accuracy Score for Year of Sale In Queens: 0.7499656357388316


In [17]:
Xblr3, Xbtestlr3, yblr3, ybtestlr3=train_test_split(brooklyn[['sale_price_clean','gross_square_feet_c']].values, brooklyn['year'].values, random_state=5)

In [18]:
clf.fit(Xblr3, yblr3)
print('Accuracy Score for Accuracy Score for Year of Sale In Brooklyn:', accuracy_score(clf.predict(Xbtestlr3), ybtestlr3))

Accuracy Score for Accuracy Score for Year of Sale In Brooklyn: 0.7606691703653123


The model is much better at predicting the year a sale took place in.  The year of a sale was correctly predicted at around 75% for both boroughs.

In [19]:
import statsmodels.api as sm

ImportError: cannot import name 'ss'